src/Engine/Objects/CombinedSkillValue.h

```diff
@@ -110,4 +111,12 @@ class CombinedSkillValue {
 
     static bool isLevelValid(int level);
     static bool isMasteryValid(PLAYER_SKILL_MASTERY mastery);
+
+    explicit operator bool() const { return _level > 0; }
+    void learn();
+    void reset();
+
+    void subtract(const int sub);
+    void add(const int add);
```

Comment:
These methods are only used in scripts. Let's not expose them in this interface, it's supposed to be type-safe. I suggest we move them somewhere closer to scripts as free-standing functions.